doc/graph/sdpa_with_compressed_kv.md

```diff
@@ -0,0 +1,124 @@
+usage, and are subsequently de-quantized to wider floating point types such as
+f16 and bf16 for computation.
+
+It's worth noting that grouped quantization is required to improve model
+accurarcy, especially for int4 data types. In this case, group size is needed
+as an attribute for quantization, which indicates the number of elements that
+share the same scaling factor and zero-points in each quantization group.
+
+The notations used in the document:
+
+- N: the mini-batch size.
+- H: the head number.
+- S: the sequence length.
+- D: the size of each head.
+- G: the group size
+
+## Pattern
+
+The SDPA with compressed Key and Value is defined as a directional acyclic graph
+(DAG) using oneDNN Graph API. oneDNN extends
+[SDPA pattern](@ref dev_guide_graph_sdpa) to support the following three kinds
+of compressed SDPA patterns:
+
+1. SDPA with compressed Key and Value.
+2. SDPA with floating-point Key and compressed Value.
+3. SDPA with compressed Key and floating-point Value.
+
+The floating-point data types includes f32, f16 and bf16, and the compressed
+data type refers to low-precision integral data types, including int4( u4/s4 )
+and int8( u8/s8 ) data types.
+
+In oneDNN Graph API, we support quantization through pattern with quantization
+operations such as `DynamicDequantize` and `DynamicQuantize`. The supported
+pattern is as follows. The blue nodes are required while the brown nodes are
+optional.
+
+![compressed SDPA pattern](images/compressed_sdpa_pattern.png)
+
+Compared to a typical SDPA pattern, there are a few differences:
+
+1. Two additional DynamicDequantize oeprations are applied to the input Key and
+Value to convert the integral cache to floating-point values. 
+2. Apart from the Query, Key and Value inputs, the pattern requires additional
+quantization information such as scale and zero-points for the dequantization of
+Key and Value caches. Currently, oneDNN only supports grouped quantization
+on one dimenstion; specifically, the shapes of scale and zero-points for Key and
+Value de-quantization should be ( N, H, S, D/G).
+3. Additionally, the `group_shape` attribute of the quantization operations must
+be specified as (1, 1, 1, G) for Key and Value dequantizaiton.
+
+## Data Types
+
+oneDNN supports the following combinations of data types for Query, Key, Value,
+output, scale for Key( scale_K ), zero-points for Key( zp_K ), scale for
+Value( scale_V ) and zero-points for Value( zp_V ):
+
+| Query   |  Key    | Scale_K | Zp_K            |  Value | Scale_V | Zp_V            | Output |
+|:--------|:--------|:--------|:----------------|:-------|:--------|:----------------|:-------|
+| dt_fp   | dt_int  | dt_fp   | u4,s4,u8,s8,s32 | dt_int | dt_fp   | u4,s4,u8,s8,s32 | dt_fp  |
+| dt_fp   | dt_int  | dt_fp   | u4,s4,u8,s8,s32 | dt_fp  | N/A     | N/A             | dt_fp  |
+| dt_fp   | dt_fp   | N/A     | N/A             | dt_int | dt_fp   | u4,s4,u8,s8,s32 | dt_fp  |
+
+Notes:
+- dt_fp can be either: f16, bf16 and f32.
```

Comment:
what's the problem if we directly expand dt_fp as `f32, bf16, f16` in the table above?

Reply:
If we directly use `f32, bf16, f16` in the table, it would result in an excessively lengthy table. Furthermore, it could lead users to misinterpret that these data types can be mixed, such as using f16 for Query with bf16 for Key scale. However, all sections involving `dt_fp` should utilize the same data type.